### DISCUSSION TITLE
Categorical data & edge and node types

### DISCUSSION THREAD

**Discussion by geoHeil** [stellargraph/stellargraph#1875] [Q&A]
Are categorical attributes on edges or nodes supported for the graph models of stellargraph?

How about support for different edge or node types? I have found: https://stellargraph.readthedocs.io/en/stable/hinsage.html which looks like a not implemented draft - and would be curious if something along these lines was implemented and how other methods would handle different node types (i.e. movies and actors) and different relation types (acted in, directed)

**Comment by roger-yu-ds** [comment]
I would also like to know this. Seems like only numerical features are supported. 

So how does one include gender in a person node? Seems like only 1a and 3 are possible. 3 is more robust because it would properly encode more than 2 categories, whereas 1a would impose an nonsensical order to unordered categories.

![image](https://user-images.githubusercontent.com/7835508/140823438-3a58e47c-f8a5-47a0-be58-740fa421fb7d.png)
